File: firebase-api-library/src/main/java/com/fazemeright/library/api/DatabaseStore.java
package com.fazemeright.library.api;

import androidx.annotation.Nullable;
import com.fazemeright.library.listeners.OnTaskCompleteListener;
import java.util.List;
import java.util.Map;
import javax.annotation.Nonnull;

public interface DatabaseStore {

  void storeUserData(@Nonnull String uid, @Nonnull Storable userData);

  void storeMessage(@Nonnull Storable object, String currentUserUid);

  void getAllMessagesForUser(@Nonnull String currentUserUid,
                             @Nullable
                                 OnTaskCompleteListener<List<Map<String, Object>>> listener);
}

File: firebase-api-library/src/main/java/com/fazemeright/library/api/DatabaseStore.kt
package com.fazemeright.library.api

import com.fazemeright.library.api.result.Result

interface DatabaseStore {
    suspend fun storeUserData(uid: String, userData: Storable): Result<Boolean>
    suspend fun storeMessage(storable: Storable, currentUserUid: String): Result<Boolean>
    suspend fun getAllMessagesForUser(currentUserUid: String): Result<List<Map<String, Any>>>?
}